src/browser/BrowserService.cpp

```diff
@@ -1026,7 +1026,17 @@ bool BrowserService::handleURL(const QString& entryUrl, const QString& url, cons
 
     // Match the subdomains with the limited wildcard
     if (siteQUrl.host().endsWith(entryQUrl.host())) {
-        return true;
+        if (!browserSettings()->bestMatchOnly()) {
+            return true;
+        }
+
+        // Match the exact subdomain and path, or start of the path when entry's path is longer than plain "/"
+        if (siteQUrl.host() == entryQUrl.host()) {
+            if (siteQUrl.path() == entryQUrl.path()
+                || (entryQUrl.path().size() > 1 && siteQUrl.path().startsWith(entryQUrl.path()))) {
```

Comment:
Is the `==` comparison supposed to match anything other than '/'? If not, you should check for that explicitly not not. `siteQUrl.path().startsWith(entryQUrl.path())` will always be true anyway if `siteQUrl.path() == entryQUrl.path()` matches.

Reply:
The `==` comparison will ralso eturn true for example two URL's with `/justsomepage` path when the subdomains are matched. The path always begins with `/` which is why the length check is needed.

Reply:
Yeah, but that is already covered by the startsWith check isn't it? startsWith is the weaker condition.

Reply:
In that case the comparison between two identical URL's without a path fails (both paths are `/`). Without the length check entry URL's path will be always matched because site URL's path always begins with `/`.

Reply:
I was thinking more along the lines of checking of '/' explicitly. But I suppose it's fine as is.